src/couch_views/src/couch_views_fdb.erl

```diff
@@ -0,0 +1,208 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_views_fdb).
+
+-export([
+    get_update_seq/2,
+    update_view_seq/3,
+    get_seq_key/2,
+
+    clear_id_index/4,
+    set_id_index/5,
+    get_id_index/4,
+    create_id_index_key/4,
+
+    clear_map_index/5,
+    set_map_index_results/5,
+    get_map_index_key/4,
+    get_map_range_keys/3,
+    get_map_range/4,
+    unpack_map_row/3
+]).
+
+
+-define(LIST_VALUE, 0).
+-define(JSON_VALUE, 1).
+-define(VALUE, 2).
+
+
+-include_lib("fabric/src/fabric2.hrl").
+-include("couch_views.hrl").
+
+% View Build Sequence Access
+% (<db>, ?DB_VIEWS, Sig, ?VIEW_UPDATE_SEQ) = Sequence
+
+get_update_seq(Db, #mrst{sig = Sig}) ->
+    #{
+        db_prefix := DbPrefix
+    } = Db,
+
+    fabric2_fdb:transactional(Db, fun(TxDb) ->
+        Key = get_seq_key(Sig, DbPrefix),
+        Tx = maps:get(tx, TxDb),
+        case erlfdb:wait(erlfdb:get(Tx, Key)) of
+            not_found -> 0;
+            UpdateSeq -> UpdateSeq
+        end
+    end).
+
+
+update_view_seq(Db, Sig, Seq) ->
+    fabric2_fdb:transactional(Db, fun(TxDb) ->
+        #{
+            db_prefix := DbPrefix,
+            tx := Tx
+        } = TxDb,
+        SeqKey = get_seq_key(Sig, DbPrefix),
+        erlfdb:set(Tx, SeqKey, Seq)
+    end).
+
+
+get_seq_key(Sig, DbPrefix) ->
+    erlfdb_tuple:pack({?DB_VIEWS, Sig, ?VIEW_UPDATE_SEQ}, DbPrefix).
```

Comment:
here and elsewhere, we might want a level of indirection rather than a 16 byte sig value in all keys.